packages/blade/src/components/Input/BaseInput/StyledBaseInput.web.tsx

```diff
@@ -167,6 +167,7 @@ const _StyledBaseInput: React.ForwardRefRenderFunction<
       onInput={(event: React.ChangeEvent<HTMLInputElement>) => {
         handleOnInput?.({ name, value: event });
       }}
+      onClick={handleOnClick}
```

Comment:
aha right! earlier we allowed onClick only on button renders. Makes sense to add this. Can you also add similar thing in react native file? I see its missing there as well.

Reply:
Sure, need to do a brief research on ReactNative, never used it.

Reply:
added the test for ReactNative, which is failing as it is not able to receive onClick prop. 
Tried some approaches like wrapping StyledNativeBaseInput  with `TouchableOpacity <TouchableOpacity  
      onPress={(): void => {
        onClick?.({ name, value: props.value }); 
      }}
    >
      <StyledNativeBaseInput`, but lot of the other tests are failing.
And how do I render TextInput as Button? I thought to add the test at least for the Button renders , but there is no prop such as 'as' to mention the type as 'button' or 'input'.

Reply:
I think we can pass it down to `onFocus` of TextInput 

https://stackoverflow.com/a/54708179

I don't think so there is any other clean way of achieving this on React Native 😢

Reply:
Ohh, got it.

Reply:
Hey @saurabhdaware @anuraghazra Not really sure how to proceed ahead on this issue for React native. Would appreciate some specific steps to resolve the issue. And from the original issue description, the second use case for onFocus+keyboard seems to be handled. Thanks.

Reply:
@rk9595 We have [StyledBaseInput.native.tsx](https://github.com/razorpay/blade/blob/48a4d4aaaf73840f00932e4f7d190d0102b642e9/packages/blade/src/components/Input/BaseInput/StyledBaseInput.native.tsx) file as well which would change things fir react native.

You can follow our [contributing guide](https://github.com/razorpay/blade/blob/master/CONTRIBUTING.md#contributing-guidelines-for-blade) to setup react native dev environment locally.

Reply:
Hi @rk9595, sorry for the late review. We were also not sure of the solution so had to pull and figure out few changes. I have made a PR to your fork with additional changes for react native and tests for it. You can merge those changes and they will get reflected in this PR - https://github.com/rk9595/blade/pull/1